app.py

```diff
@@ -0,0 +1,31 @@
+from abc import ABC, abstractmethod
+import speech_recognition as sr
+
+# Abstract class that defines the methods amd attributes of Braille Apps. 
+class App(ABC):
+
+    def __init__(self, cells, audio):
+        self.cells = cells
+        self.audio = self.audio
```

Comment:
I'm less familiar with abstract classes in Python, but each app needs to have it's own name, which I think should be defined here.

Is there a way we can force the initialisation of a self.name (or similarly-named) field?

Reply:
Added name to constructor.